ignite/metrics/fid.py

```diff
@@ -0,0 +1,131 @@
+import os
+from typing import Callable, Sequence, Union
+
+import torch
+from scipy import linalg
+from torchvision import transforms
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+
+__all__ = ["FID"]
+
+class DefaultDataset(data.Dataset):
+    def __init__(self, root, transform=None):
+        self.samples = listdir(root)
+        self.samples.sort()
+        self.transform = transform
+        self.targets = None
+
+    def __getitem__(self, index):
+        fname = self.samples[index]
+        img = Image.open(fname).convert('RGB')
+        if self.transform is not None:
+            img = self.transform(img)
+        return img
+
+    def __len__(self):
+        return len(self.samples)
+
+class FID(Metric):
+    """Calculates FID metric
+
+    """
+
+    def __init__(
+        self, 
+        output_transform: Callable = lambda x: x, 
+        input_path: os.path, 
+        output_path: os.path,
+        test_model = None,
```

Comment:
I like the implementation of fid here : https://github.com/Tetratrio/stylegan2_pytorch/blob/master/stylegan2/metrics/fid.py
and I think it would be better to call test_model as fid_model ?